core/rovecomm.py

```diff
@@ -73,6 +86,39 @@ def print(self):
         print('----------')
 
 
+class RoveCommCollection:
+    '''
+    Creates a separate thread to read all RoveComm connections
+    '''
+    connections = []
+    shutdown_event = None
+
+    def __init__(self):
+        self.shutdown_event = threading.Event()
+        self.thread = threading.Thread(target=self.listen)
+        self.thread.start()
+
+    def listen(self):
+        while threading.main_thread().is_alive() and not self.shutdown_event.isSet():
```

Comment:
Confirm 100% that this is the correct way of checking whether we should still be running.